diff --git a/node_modules/next/dist/server/app-render/get-segment-param.js b/node_modules/next/dist/server/app-render/get-segment-param.js
index 7a06282903722271c64188b49f99e43679f29113..6ed8e7724128f1190471c26c6d26732dc6e17424 100644
--- a/node_modules/next/dist/server/app-render/get-segment-param.js
+++ b/node_modules/next/dist/server/app-render/get-segment-param.js
@@ -51,7 +51,10 @@
         // Leverage the `use` hook to ensure that we're compatible with Suspense.
         if (typeof tree[1] === "object" && tree[1] !== null && typeof tree[1].then === "function") {
             param = (0, _react).use(tree[1]);
-        } else if (tree[1] === _constants.PAGE_SEGMENT_KEY) {
+        } else if (tree[1] === _constants.PAGE_SEGMENT_KEY ||
+            // This can happen if the file is living in `app/page.js` instead of
+            // `app/(some-group)/page.js` and the layout is wrapping it.
+            tree[0] === "__PAGE__") {
             param = undefined;
         } else {
             param = tree[1];

